### ISSUE TITLE
Sqlalchemy doesn't detect Identity columns when the table name has uppercase letters

### ISSUE THREAD

**Issue by Bechma** [sqlalchemy/sqlalchemy#6129]
**Describe the bug**
Alembic 1.5.8 with version SQLAlchemy 1.4.2 is not detecting the Identity columns when the table name has uppercase letters. I was trying to run alembic automigrations but it's always saying that I have to modify the id column that has "GENERATED BY DEFAULT AS IDENTITY" property to server_default=None.

After some trial and error I discovered that when the table name has all lower case characters, alembic works as expected and says that no migration is needed. However, if I change the table name and capitalize the names, alembic begins to say that I have to "modify_default" with an Identity property.

**Expected behavior**
The behaviour that alembic has right now with the lower case tables should be the same that the tables with upper case characters.

**To Reproduce**
```py
from alembic.autogenerate import compare_metadata
from alembic.runtime.migration import MigrationContext
from sqlalchemy import Column, Integer, Identity
from sqlalchemy.orm import declarative_base

from dbconn import engine

Base = declarative_base(bind=engine)


class Example(Base):
    """example definition table"""
    __tablename__ = "Example"

    id = Column(Integer, Identity(), primary_key=True)


Base.metadata.create_all()

with engine.connect() as conn:
    mc = MigrationContext.configure(conn)
    diff = compare_metadata(mc, Base.metadata)
    assert len(diff) == 0, f"{diff}"
```

**Error**

```
Traceback (most recent call last):
  File "C:/test.py", line 26, in <module>
    assert len(diff) == 0, f"{diff}"
AssertionError: [[('modify_default', None, 'Example', 'id', {'existing_nullable': False, 'existing_type': INTEGER(), 'existing_comment': None}, None, Identity())]]

```

**Versions.**
 - OS: Windows 10
 - Python: 3.8
 - Alembic: 1.5.8
 - SQLAlchemy: 1.4.2
 - Database: Postgresql 12.4
 - DBAPI: psycopg2 2.8.6

**Additional context**
My workaround for now is to stay with the old method of 

```py
@compiles(CreateColumn, 'postgresql')
def use_identity(element, compiler, **kw):
    text = compiler.visit_create_column(element, **kw)
    text = text.replace("SERIAL", "INT GENERATED BY DEFAULT AS IDENTITY")
    return text
```

with the tables that have upper case characters. This way alembic does not complain(although I think it should)

**Have a nice day!**

**Comment by CaselIT** [comment]
This is a bug in sqlalchemy, in the reflection select. Sadly the pg_catalog table of postgress are not very user friendly to work with.
Alas the fix is trivial.

Reproduction without alembic:
```py
from sqlalchemy import Column, Integer, Identity, Table
from sqlalchemy.orm import declarative_base
import sqlalchemy as sa
from sqlalchemy import create_engine

engine = create_engine("postgresql://scott:tiger@127.0.0.1:5432/test")
Base = declarative_base()
class A(Base):
    """example definition table"""
    __tablename__ = "Example"
    id = Column(Integer, Identity(), primary_key=True)
class B(Base):
    """example definition table"""

    __tablename__ = "example"
    id = Column(Integer, Identity(), primary_key=True)
Base.metadata.drop_all(engine)
Base.metadata.create_all(engine)
isp = sa.inspect(engine)
print(isp.get_columns('Example'))
print(isp.get_columns('example'))
```

**Comment by sqla-tester** [comment]
**Federico Caselli** has proposed a fix for this issue in the **master** branch:

**Fixed reflection of identity columns in tables** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2676

**Comment by CaselIT** [comment]
@Bechma thanks for the report, we were using a rather fragile query to reflect the identity of the columns.
The fix should be in 1.4.3 that should be out soon
